tests/antithesis/test-template/Dockerfile

```diff
@@ -1,4 +1,4 @@
-ARG GO_VERSION=1.24.2
+ARG GO_VERSION=1.24.3
```

Comment:
@serathius, should we bump this Dockerfile's version? cc. @nwnt

Reply:
It should not cause problems